asg/src/reducer/monoid/forgetful.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the Leo library.
+
+// The Leo library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The Leo library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the Leo library. If not, see <https://www.gnu.org/licenses/>.
+
+use super::*;
+
+/// This monoid ignores all append operations
+pub struct Fixed<T: Default>(pub T);
+
+impl<T: Default> Default for Fixed<T> {
+    fn default() -> Self {
+        Fixed(T::default())
```

Comment:
It does not hold that `append(T::default(), other) = other` so this is *not* a monoid.

Reply:
Oops I totally missed that! It doesn't satisfy the identity property. 
How about instead of using monoids, we use magmas? We can even use semigroups but not sure if it's important to have the restriction of associativity.

Reply:
Not sure about associativity either. Could you look through the uses and see what makes most sense?

Reply:
We'll go with magmas since they are more general and because we are not enforcing the associative/identity properties of monoids. The reducer also does not seem to rely on any of these properties.